Simbody/Visualizer/include/simbody/internal/Visualizer.h

```diff
@@ -760,6 +764,44 @@ class SimTK_SIMBODY_EXPORT Visualizer::FrameController {
     virtual ~FrameController() {}
 };
 
+/** Causes the camera to point at and follow a point fixed on a body (a
+ station). This might be useful if your system translates substantially
+ and would thus leave the field of view of a stationary camera. **/
+// This class is based on the BodyWatcher class that used to be in TimsBox.cpp.
+class SimTK_SIMBODY_EXPORT Visualizer::BodyFollower :
+    public Visualizer::FrameController {
+public:
+    /**
+    @param[in] bodyB
+       The body to follow.
+    @param[in] stationPinB
+       The location of the station P on the body to follow, expressed in B.
+    @param[in] offset
+       Position of the camera from P, expressed in ground.
+    @param[in] upDirection
+       Controls the rotation of the camera about the offset vector. The
+       camera's up (+y) direction will be aligned with this vector as best as
+       possible. Expressed in ground. By default, this is the system's up
+       direction (see getSystemUpDirection); it's unlikely that you want
+       something other than the default.
+    **/
+    BodyFollower(const MobilizedBody& body,
+                          const Vec3&          stationPinB = Vec3(0, 0, 0),
+                          const Vec3&          offset = Vec3(NaN),
+                          const Vec3&          upDirection = Vec3(NaN));
+    
+    virtual void generateControls(
+            const Visualizer&             viz,
+            const State&                  state,
+            Array_< DecorativeGeometry >& geometry) override;
+    
+private:
+    const MobilizedBody& m_body;
+    const Vec3 m_stationPinB;
+    const Vec3 m_offset;
+    const Vec3 m_upDirection;
```

Comment:
Style comment: you might consider aligning all the data members here, like:
```
    const MobilizedBody& m_body
    const Vec3           m_stationPinB;
    const Vec3           m_offset;
    const Vec3           m_upDirection;
```
I think it is marginally easier to scan that way, but feel free to ignore.